mysk-data-api/src/routes/v1/subjects/electives/modify_electives.rs

```diff
@@ -83,28 +88,38 @@ async fn modify_elective_subject(
         Err(Error::InternalSeverError(_, _)) => {
             return Err(Error::InvalidRequest(
                 "Elective subject not found".to_string(),
-                format!("/subjects/electives/{session_code}/enroll"),
+                format!("/subjects/electives/{elective_subject_session_id}/enroll"),
             ));
         }
         _ => unreachable!("ElectiveSubject::get_by_id should always return a Detailed variant"),
     };
 
     // Checks if the student is in a class available for the elective
-    if !DbElectiveSubject::is_student_eligible(pool, session_code, student_id).await? {
+    if !DbElectiveSubject::is_student_eligible(pool, elective_subject_session_id, student_id)
+        .await?
+    {
         return Err(Error::InvalidPermission(
             "Student is not eligible to enroll in this elective".to_string(),
-            format!("/subjects/electives/{session_code}/enroll"),
+            format!("/subjects/electives/{elective_subject_session_id}/enroll"),
         ));
     }
 
     // Checks if the student has already enrolled in the elective before
+    let subject_id = DbElectiveSubject::get_by_id(pool, elective_subject_session_id)
+        .await?
+        .subject_id;
+
     let enroll_count = query!(
-        r#"
-        SELECT COUNT(*) FROM student_elective_subjects
-        WHERE student_id = $1 AND elective_subject_id = $2
-        "#,
+        r"
+        SELECT 
+            COUNT(*) 
+        FROM 
+            elective_subject_session_enrolled_students 
+            INNER JOIN elective_subject_sessions_with_detail_view on elective_subject_session_enrolled_students.elective_subject_session_id = elective_subject_sessions_with_detail_view.id
+        WHERE student_id = $1 AND subject_id = $2
```

Comment:
Refer to the first comment in `enroll_electives.rs`.

Reply:
Took a look at the comments, we're checking for their history so yours is already right